client/python/cryoet_data_portal/src/cryoet_data_portal/_codegen.py

```diff
@@ -0,0 +1,280 @@
+"""This module generates model code from the GraphQL schema."""
+
+import logging
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+from gql import Client
+from gql.transport.requests import RequestsHTTPTransport
+from graphql import (
+    GraphQLEnumType,
+    GraphQLField,
+    GraphQLList,
+    GraphQLNamedType,
+    GraphQLNonNull,
+    GraphQLObjectType,
+    GraphQLScalarType,
+    GraphQLSchema,
+    GraphQLType,
+    build_schema,
+    print_schema,
+)
+from jinja2 import Environment, FileSystemLoader
+
+from cryoet_data_portal._client import DEFAULT_URL
+
+"""Maps GraphQL field type names to model field defaults and Python types."""
+GQL_TO_MODEL_FIELD = {
+    "Boolean": ("BooleanField()", "bool"),
+    "Float": ("FloatField()", "float"),
+    "Int": ("IntField()", "int"),
+    "String": ("StringField()", "str"),
+    "date": ("DateField()", "date"),
+    "numeric": ("FloatField()", "float"),
+    "_numeric": ("StringField()", "str"),
+    "tomogram_type_enum": ("StringField()", "str"),
+    # TODO: check if I should make a JsonField to handle this.
+    "json": ("StringField()", "str"),
```

Comment:
Unsure how this value should be parsed into `Annotation.method_links`.

Should this be a `str` (as now) or a `dict`? I still need to look at real live values to understand this.

Reply:
This would be a `dict`, as it would be a JsonField. :)

Reply:
I removed it for now (so it's ignored) since @jgadling was unsure if this should be live and public. My plan is to add it in a follow-up PR since this one is more than large enough!